frontend/src/components/course/course.tsx

```diff
@@ -1,29 +1,50 @@
 import defaultCourseImage from 'assets/img/default-course-image.jpeg';
-import { DataStatus } from 'common/enums/enums';
+import { DataStatus, PermissionKey } from 'common/enums/enums';
 import { FC } from 'common/types/types';
-import { Content, Image, Spinner } from 'components/common/common';
+import { Category, Content, Image, Spinner } from 'components/common/common';
+import { checkHasPermission } from 'helpers/helpers';
 import {
   useAppDispatch,
   useAppSelector,
   useEffect,
   useParams,
+  useState,
 } from 'hooks/hooks';
 import { courseActions } from 'store/actions';
 
+import { AddUpdateCategoryModal } from './components/components';
+import { EditButton } from './components/edit-button/edit-button';
 import { ModulesCardsContainer } from './components/modules-cards-container/modules-cards-container';
 import styles from './styles.module.scss';
 
 const Course: FC = () => {
-  const { course, modules, dataStatus } = useAppSelector(
+  const { course, modules, categories, dataStatus } = useAppSelector(
     (state) => state.course,
   );
+  const { user } = useAppSelector((state) => state.auth);
   const { id } = useParams();
   const dispatch = useAppDispatch();
 
+  const categoryIsAllowedToEdit = checkHasPermission({
```

Comment:
boolean variable should be named as an answer on general question (B7)
```suggestion
  const isCategoryEditAllowed= checkHasPermission({
```

Reply:
Oh, sorry. I interpreted that rule in way that the place of word "is" or "has" is not fundamental. I will fix that!